rstudio/rhel9-python-3.9/Dockerfile

```diff
@@ -24,7 +20,11 @@ USER root
 #RUN sed -i 's/\(def in_container():\)/\1\n    return False/g' /usr/lib64/python*/*-packages/rhsm/config.py
 
 # Run the subscription manager command using the provided credentials. Only include --serverurl and --baseurl if they are provided
-RUN subscription-manager register \
+RUN SERVERURL=$(cat ${SECRET_DIR}/SERVERURL) && \
+    BASEURL=$(cat ${SECRET_DIR}/BASEURL) && \
+    USERNAME=$(cat ${SECRET_DIR}/USERNAME) && \
+    PASSWORD=$(cat ${SECRET_DIR}/PASSWORD) && \
+    subscription-manager register \
     ${SERVERURL:+--serverurl=$SERVERURL} \
     ${BASEURL:+--baseurl=$BASEURL} \
     --username=$USERNAME \
```

Comment:
I get

```
STEP 5/29: RUN SERVERURL=$(cat ${SECRET_DIR}/SERVERURL) &&     BASEURL=$(cat ${SECRET_DIR}/BASEURL) &&     USERNAME=$(cat ${SECRET_DIR}/USERNAME) &&     PASSWORD=$(cat ${SECRET_DIR}/PASSWORD) &&     subscription-manager register     ${SERVERURL:+--serverurl=$SERVERURL}     ${BASEURL:+--baseurl=$BASEURL}     --username=$USERNAME     --password=$PASSWORD     --force     --auto-attach
cat: /opt/app-root/src/.sec/SERVERURL: No such file or directory
```

I am specifying a secret with only username and password in it,

```
oc create secret generic rhel-subscription-secret --from-literal=USERNAME=<username> --from-literal=PASSWORD=<password> -n redhat-ods-applications
```

Reply:
So it seems that we have two options here:

Either to specify the baseurl and serverurl with empty string like :
```oc create secret generic rhel-subscription-secret --from-literal=USERNAME=xxxx --from-literal=PASSWORD=yyyy --from-literal=SERVERURL='' --from-literal=BASEURL='' ```

Or Remove them (the baseurl and serverurl) from the dockerfile?

Reply:
or do sophisticated bash-scripting here, which I never believed is something I'll end up advocating, but here we go

```
BASEURL=$(cat ${SECRET_DIR}/BASEURL 2>&-)
```

I have no idea how it works, but StackOverflow told me to do it, https://stackoverflow.com/questions/53859603/bash-one-liner-for-if-a-file-exists-read-it-into-variable-else-empty-string

Reply:
I am checking this as well, ty!

Reply:
Thanks folks, while you were suggesting, I was implementing this way. 
I used defaults values for server and baseurl, like `ARG SERVERURL_DEFAULT=""
ARG BASEURL_DEFAULT=""` and on the subscription command if there is not specified value, to use the default, like: 
`SERVERURL=$(cat ${SECRET_DIR}/SERVERURL 2>/dev/null || echo ${SERVERURL_DEFAULT})`.

@Jiri once you have time could you take a look?

Reply:
That's a different Jiri you tagged. Sorry Jiri, it was for me. ;P

Reply:
@atheo89 this should work; I'll go check if I get an itch, but the code looks right to me; I'm going to finish my day trying to spawn some openshifts for tomorrow's live testing, I guess

Reply:
Thanks @atheo89 @jiridanek @jstourac for the discussion.
you feel satisfied, please leave your review with lgtm or otherwise.